packer/provisioner.go

```diff
@@ -145,17 +145,65 @@ func (p *PausedProvisioner) Prepare(raws ...interface{}) error {
 
 func (p *PausedProvisioner) Provision(ctx context.Context, ui packersdk.Ui, comm packersdk.Communicator, generatedData map[string]interface{}) error {
 
-	// Use a select to determine if we get cancelled during the wait
 	ui.Say(fmt.Sprintf("Pausing %s before the next provisioner...", p.PauseBefore))
-	select {
-	case <-time.After(p.PauseBefore):
-	case <-ctx.Done():
-		return ctx.Err()
+	err := p.pause(ctx, ui)
+	if err != nil {
+		return err
 	}
 
 	return p.Provisioner.Provision(ctx, ui, comm, generatedData)
 }
 
+func (p *PausedProvisioner) pause(ctx context.Context, ui packersdk.Ui) error {
+	if p.PauseBefore == time.Duration(0) {
+		return nil
+	}
+
+	updateTime := 10
```

Comment:
Do you think it would be a good idea to make this an option so the everyone gets an opt-in ?

Reply:
I'm thinking of allowing to set the `tick_time = "10s"` ( and if it's zero then we don't tick ).

Reply:
Added TickSeconds to PausedProvisioner for this feature.

Reply:
@azr Any update on this?